services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java

```diff
@@ -125,8 +128,22 @@ public Expression filterExpression() {
 
     /**
      * Returns the list of projected attributes on this request object, or an null if no projection is specified.
+     * The Nested Attributes are represented using DOT separator.
+     * Example : foo.bar is represented as "foo.bar" which is indistinguishable from a non-nested attribute
+     * with the name "foo.bar".
+     * Use {@link #nestedAttributesToProject} if you have a use-case that requires discrimination between these two cases.
      */
     public List<String> attributesToProject() {
+        return attributesToProject != null ? attributesToProject.stream().filter(Objects::nonNull)
+                .map(item -> String.join(".", item.elements())).collect(Collectors.toList()) : null;
```

Comment:
I feel like we should add validation to the constructor of the class to avoid the possibility of having null elements in attributesToProject then there would be no need to filter them here.